### PR TITLE
Binder resources

### PR DESCRIPTION
**Pull request by karenc**
- Add resources to the binder model

  - Output list of resources in metadata in EPUB html
    
    This allows us to attach resources back to the correct models when we
    parse the EPUB, including resources that are not used in the html.
    (e.g. The cover image for a book)

  - Parse and attach binder resources from EPUB
    
    Resource names are stored in the binder HTML metadata
    data-type="resources" div.

### CODE REVIEW

**Comment by philschatz** [comment]
:+1:

**Comment by pumazi** [comment]
Looks good. :+1:

One change... sorry, my fault. :confounded: Can we remove the ``class="resources"``?

**Comment by karenc** [comment]
No problem :)  Removed `class="resources"`

```diff
diff --git a/cnxepub/adapters.py b/cnxepub/adapters.py
index f7e5110..d077dd0 100644
--- a/cnxepub/adapters.py
+++ b/cnxepub/adapters.py
@@ -606,7 +606,7 @@ HTML_DOCUMENT = """\
       <div itemprop="about" data-type="subject">{{ subject }}</div>
       {%- endfor %}
 
-      <div class="resources" data-type="resources" style="display: none">
+      <div data-type="resources" style="display: none">
         <ul>
           {% for resource in resources -%}
           <li>{{ resource.id }}</li>
diff --git a/cnxepub/tests/data/book/content/9b0903d2-13c4-4ebe-9ffe-1ee79db28482@1.6.
index 27693a0..fb4f665 100755
--- a/cnxepub/tests/data/book/content/9b0903d2-13c4-4ebe-9ffe-1ee79db28482@1.6.xhtml
+++ b/cnxepub/tests/data/book/content/9b0903d2-13c4-4ebe-9ffe-1ee79db28482@1.6.xhtml
@@ -148,7 +148,7 @@
            data-type="subject"
            >Science and Mathematics</div>
 
-      <div class="resources" data-type="resources" style="display: none">
+      <div data-type="resources" style="display: none">
         <ul>
           <li>cover.png</li>
         </ul>
diff --git a/cnxepub/tests/data/book/content/e78d4f90-e078-49d2-beac-e95e8be70667@3.xh
index 93e8471..fdcfeee 100755
--- a/cnxepub/tests/data/book/content/e78d4f90-e078-49d2-beac-e95e8be70667@3.xhtml
+++ b/cnxepub/tests/data/book/content/e78d4f90-e078-49d2-beac-e95e8be70667@3.xhtml
@@ -164,7 +164,7 @@
            data-type="subject"
            >Science and Mathematics</div>
 
-      <div class="resources" data-type="resources" style="display: none">
+      <div data-type="resources" style="display: none">
         <ul>
           <li>e3d625fe893b3f1f9aaef3bdf6bfa15c.png</li>
         </ul>
```

File: cnxepub/models.py [hunk 1]
```diff
@@ -337,12 +337,14 @@ def insert(self, i, v):
 
 class Binder(TranslucentBinder):
     """An object that has metadata and contains
-    ``Binder``, ``TranslucentBinder`` and ``Document`` instances.
+    ``Binder``, ``Resource``, ``TranslucentBinder`` and ``Document`` instances.
```
Comment (pumazi) [inline cnxepub/models.py]: Technically, it doesn't "contain" Resources the same way. In this we are talking about the containment as in contained within a sequence and therefore useful for the ``__contains__`` method.

**Pull request merged**